include/item.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "ranges.h"
+
+namespace celerity {
+
+template <int Dims>
+class item;
+
+namespace detail {
+
+	template <int Dims>
+	item<Dims> make_item(cl::sycl::id<Dims> global_id, cl::sycl::range<Dims> global_range) {
+		return item<Dims>{global_id, global_range};
+	}
+
+} // namespace detail
+
+// We replace sycl::item with celerity::item to correctly expose the cluster global size instead of the chunk size to the user.
+template <int Dims>
+class item {
+  public:
+	item() = delete;
+
+	cl::sycl::id<Dims> get_id() const { return id; }
+
+	size_t get_id(int dimension) const { return id[dimension]; }
+
+	operator cl::sycl::id<Dims>() const { return id; } // NOLINT(google-explicit-constructor)
+
+	size_t operator[](int dimension) const { return id[dimension]; }
+
+	cl::sycl::range<Dims> get_range() const { return range; }
+
+	size_t get_range(int dimension) const { return range[dimension]; }
+
+	size_t get_linear_id() const { return detail::get_linear_index(range, id); }
```

Comment:
As just discussed in person, we should probably adjust the behavior of `get_linear_id` to that prescribed by SYCL 2020, i.e., subtracting the kernel offset before linearizing, such that the first element always has a linear id of 0.

Reply:
Adjusted, also added `get_offset()` from `sycl::item`. This now requires us to include an additional `sycl::id<Dims> global_offset`. I'm not entirely convinced we should keep global offsets around indefinitely since SYCL 2020 has now deprecated them.

Reply:
Agreed. And we probably don't need to wait until SYCL Next to do so (but we should at least have a grace period with deprecation warning). I'll add it to *the list*.